src/app/(tools)/coverLetter/page.tsx

```diff
@@ -0,0 +1,285 @@
+"use client";
+
+import { useState, useRef } from 'react';
+import { Upload, X } from 'lucide-react';
+import axios from 'axios';
+import jsPDF from 'jspdf';
+import { useForm, Controller } from 'react-hook-form';
+import { z } from 'zod';
+import { zodResolver } from '@hookform/resolvers/zod';
+
+import { Button } from "@/components/ui/button";
+import { Textarea } from "@/components/ui/textarea";
+import { toast } from '@/hooks/use-toast';
+
+const schema = z.object({
+  jobDescription: z.string().min(1, "Job description cannot be empty"),
+  resume: z.string().min(1, "Resume cannot be empty")
+});
+
+type FormData = z.infer<typeof schema>;
+
+export default function Page() {
+  const [coverLetter, setCoverLetter] = useState('');
+  const [loading, setLoading] = useState(false);
+  const [uploadedFileName, setUploadedFileName] = useState<string | null>(null);
+  const fileInputRef = useRef<HTMLInputElement>(null);
+  const coverLetterRef = useRef<HTMLTextAreaElement>(null);
+
+  const { control, handleSubmit, setValue, formState: { errors } } = useForm<FormData>({
+    resolver: zodResolver(schema),
+    defaultValues: {
+      jobDescription: '',
+      resume: ''
+    }
+  });
+
+  const handleResumeChange = (e: React.ChangeEvent<HTMLTextAreaElement>) => {
+    setValue('resume', e.target.value);
+  };
+
+  const handleResumePaste = (e: React.ClipboardEvent) => {
+    const items = e.clipboardData.items;
+    for (let i = 0; i < items.length; i++) {
+      if (items[i].kind === 'file') {
+        e.preventDefault();
+        const file = items[i].getAsFile();
+        if (file) handleFileUpload(file);
+        break;
+      }
+    }
+  };
+
+  const handleFileUpload = async (file: File) => {
+    setLoading(true);
+    setUploadedFileName(file.name);
+    try {
+      const base64File = await fileToBase64(file);
+      const fileType = file.type;
+      const response = await axios.post(`${process.env.NEXT_PUBLIC_BACKEND}/api/parse-resume`, {
```

Comment:
this is soo 2015's way of calling api's and handling its state.

use https://tanstack.com/query/v3 and you'll see it will make the code cleaner automatically